client/component/src/DomainConfig.tsx

```diff
@@ -86,8 +86,24 @@ interface DomainConfigProps {
 function DomainConfig(props: DomainConfigProps) {
   const { dataDomain, customDomain, scaleType } = props;
   const { onCustomDomainChange, histogram } = props;
+  const [lockDomain, setLockDomain] = useState<boolean>(false);
 
   const visDomain = useVisDomain(customDomain, dataDomain);
+
+  const memoizedCustomDomain = useMemo(() => {
+    if (lockDomain) {
+      return customDomain;
+    } else {
+      return visDomain;
+    }
+  }, [lockDomain, customDomain, visDomain]);
+
+  useEffect(() => {
+    if (!lockDomain) {
+      onCustomDomainChange(dataDomain);
+    }
+  }, [dataDomain, onCustomDomainChange]);
```

Comment:
missing `lockDomain`

Reply:
If lockDomain is included, then domain resets when the lockDomain checkbox is unchecked.

Reply:
This is because the setter below is wrong. State variables are lazy and don't update straightaway. Read https://react.dev/reference/react/useState for a thorough understanding.

Reply:
Updated setter